internal/credentials/kubernetes/github/app_test.go

```diff
@@ -48,8 +48,26 @@ func TestAppCredentialHelper(t *testing.T) {
 				require.Nil(t, creds)
 			},
 		},
+		{
+			name:     "cred type is git",
+			credType: credentials.TypeGit,
+			helper:   &appCredentialHelper{},
+			assertions: func(t *testing.T, creds *credentials.Credentials, _ *cache.Cache, err error) {
+				require.NoError(t, err)
+				require.Nil(t, creds)
+			},
+		},
```

Comment:
I think the number of new test cases is not entirely necessary. We mainly utilize "white-box testing," which means the tests are written with enough awareness of the implementation such that the test cases are engineered to force execution down as many paths in the code as possible. The flip side of this is that it sometimes reduces the number of cases required to test the code thoroughly. In this particular situation, we know that `credType` isn't used anywhere other than in the one conditional you have updated -- meaning that once that conditional is cleared, the value of `credType` has no further bearing on the outcome of any given test case. i.e. Test cases `no github details provided` and `no github details provided - git` are effectively redundant.

I believe, where tests are concerned, you only need two small changes _instead of all these new cases_: Change the name `cred type is not git` to `cred type is not supported` and in that test case, change `credType: credentials.TypeGit,` to `credType: credentials.TypeHelm`.

Note: GHCR _can_ store Helm charts, meaning a GitHub App _should_ be a viable authentication method for a Helm chart stored in GHCR, but I think that can be dealt with in a follow-up PR because it is slightly more involved than this change was.